middleware/validate.js

```diff
@@ -0,0 +1,168 @@
+          return res.render('./articles/new', { message: "That article is not in our database, please add it as a new article" });
+        }
+      }
+
+      // POST '/articles'
+      if (req.method === 'POST' && req.url === '/articles') {
+        let body = req.body;
+
+        if (Object.keys(body).length < 3) {
+          return res.render('./products/new', { message: "Please fill out all fields with approppriate information" });
+        }
+
+        for (var key in body) {
+          if (body[key].length < 2) {
+            return res.render('./products/new', { message: "Please fill out all fields with approppriate information" });
+          }
+
+          if (!isNaN(parseInt(body[key]))) {
+            return res.render('./products/new', { message: "Please fill out all fields with approppriate information" });
+          }
+        }
+      }
+
+      // PUT '/:title'
+      if (req.method === "PUT" && req.url === `/${title}`) {
+        let body = req.body;
+        const params = req.params
+        const articleIndex = articleDB.findArticle(params.title);
+
+        if (typeof articleIndex !== 'number') {
+          let data = { message: "That article is not in our database, please add it as a new article" };
+
+          return res.render('./articles/new', data);
+        }
+
+        for (var key in body) {
+          if (body[key].length < 2 || !isNaN(parseInt(body[key]))) {
+            return res.render(`./products/edit`, {
+              id: params.id,
+              title: database[articleIndex].title,
+              author: database[articleIndex].price,
+              body: database[articleIndex].body,
+              message: "Please fill out all fields with approppriate information"
+            });
+          }
+        }
+      }
+
+      // DELETE '/:title'
+      if (req.method === "DELETE" && req.url === `/${title}`) {
+        const articleTitle = req.params.title;
+        const articleIndex = articleDB.findArticle(articleTitle);
+
+        if (typeof articleIndex !== 'number') {
+          res.status(500);
+          return res.json({ "Error": "That product does not exist in our database." })
+        }
+      }
+    }
+
+
+    /************************
+    *  PRODUCTS
+    ************************/
+    const id = req.params.id;
+    const productDB = require('../db/products.js');
+    const database = productDB.getProducts().products;
+
+    // GET '/:id'
+    if (req.method === 'GET' && req.url === `/${id}`) {
+      const params = req.params;
+      const productIndex = productDB.findProduct(params.id);
+
+      if (typeof productIndex !== 'number') {
+        res.status(500);
+        return res.json({ "Error": "Couldn't find that product in our database." });
+      }
+    }
+
+    // GET '/:id/edit'
+    if (req.method === "GET" && req.url === `/${id}/edit`) {
+      let params = req.params;
+      let productIndex = productDB.findProduct(params.id);
+      let product = database[productIndex];
```

Comment:
You shouldn't try to use the array inside productDb like this. You should stick with the methods expose through the module.exports.